src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -769,6 +771,252 @@ protected bool IsAlive
             }
         }
 
+        /// <summary>
+        /// Gets the child Elements of the current Element.
+        /// </summary>
+        /// <returns>Child Elements.</returns>
+        public IEnumerable<Element> GetChildElements()
+        {
+            return GetElementChildElements(this.InternalElement);
+            
+        }
+
+        private IEnumerable<Element> GetElementChildElements(Autodesk.Revit.DB.Element element)
+        {
+            List<Element> components = new List<Element>();
+            BuiltInCategory builtInCategory = (BuiltInCategory)System.Enum.Parse(typeof(BuiltInCategory),
+                                                                                 element.Category.Id.ToString());
+
+            switch (builtInCategory)
+            {
+                case BuiltInCategory.OST_Stairs:
+                    List<Element> stairComponentElements = GetChildElementsFromStairs(element);
+                    components.AddRange(stairComponentElements);
+                    break;
+
+                case BuiltInCategory.OST_StructuralFramingSystem:
+                    List<Element> beamSystemComponentElements = GetChildElementsFromStructuralFramingSystem(element);
+                    components.AddRange(beamSystemComponentElements);
+                    break;
+
+                case BuiltInCategory.OST_StairsRailing:
+                    List<Element> railingComponentElements = GetChildElementsFromRailings(element);
+                    components.AddRange(railingComponentElements);
+                    break;
+
+                default:
+                    List<Element> componentElements = GetChildElementsFromFamilyInstance(element);
+                    components.AddRange(componentElements);
+                    break;
+            }
+            return components;
+        }
+
+        private static List<Element> GetChildElementsFromFamilyInstance(Autodesk.Revit.DB.Element element)
+        {
+            var familyInstance = element as Autodesk.Revit.DB.FamilyInstance;
+            if (familyInstance == null)
+                throw new NullReferenceException(Properties.Resources.NoChildElements);
```

Comment:
throwing and catching exceptions is quite slow - are we sure we don't just want to return an empty list?

Reply:
@mjkkirschner I guess we are trying to distinguish the case that an element comes with zero child elements V.S. an element is simply not supported for such node operation?

Reply:
our thought on throwing is to provide more information to the user, might be confusing to just get an empty list as it might not be if something was done wrong